File: front-end/auctions-web/app/js/services/message.service.js
angular.module('auctions.message')

.service(
		"MessageService",
		function($resource, SERVER_IP_ADDRESS) {
			return $resource('http://' + SERVER_IP_ADDRESS.ipAddress
					+ '/auctions/messages/:messageId', {
				messageId : '@messageId'
			}, {
				get : {
					method : 'GET'
				},
				delete : {
					method : 'DELETE'
				},
				save : {
					method : 'POST',
					transformRequest : function(data, headers) {
						headers = angular.extend({}, headers, {
							'Content-Type' : 'application/json'
						});
						console.log(data);
						console.log(angular.toJson(data));
						return angular.toJson(data); // this will go in the body request
					}
				},

			});

		});

File: back-end/auctions-source/src/main/java/app/auctions/service/MessageService.java
package app.auctions.service;

import java.util.List;

import app.auctions.model.Message;

public interface MessageService {
	public void save(Message message);
	public int delete(long messageId);
	public void deleteFromInbox(long messageId);
	public void deleteFromOutbox(long messageId);
	public List<Message> findInboxedMessagesByReceiver(long userId);
	 public List<Message> findOutboxedMessagesBySender(long userId);
	 public Message findMessageById(long messageId);
}
